domain_menu_access.module

```diff
@@ -53,39 +53,14 @@ function domain_menu_access_form_menu_link_content_form_alter(&$form, FormStateI
 }
 
 /**
- * Implements hook_form_alter().
+ * Implements hook_block_alter().
  */
-function domain_menu_access_form_domain_access_settings_alter(&$form, FormStateInterface $form_state, $form_id) {
-  $config = \Drupal::config('domain_access.settings');
-
-  /** @var \Drupal\Core\Entity\EntityStorageInterface $menu_storage */
-  $menu_storage = \Drupal::service('entity.manager')->getStorage('menu');
-
-  $options = [];
-  /** @var \Drupal\system\Entity\Menu $menu */
-  foreach ($menu_storage->loadMultiple() as $menu) {
-    $options[$menu->getOriginalId()] = $menu->get('label');
+function domain_menu_access_block_alter(&$definitions) {
+  foreach ($definitions as $id => $definition) {
+    if (strpos($id, 'system_menu_block:') === 0) {
+      $definition['id'] = 'domain_access_menu_block';
+      $definition['deriver'] = 'Drupal\domain_menu_access\Plugin\Derivative\DomainMenuAccessMenuBlock';
+      $definition['class'] = 'Drupal\domain_menu_access\Plugin\Block\DomainMenuAccessMenuBlock';
+    }
   }
-  asort($options, SORT_NATURAL);
-
-  $form['menu_enabled'] = array(
-    '#type' => 'checkboxes',
-    '#title' => t('Enable Domain Access fields on menu link contents:'),
-    '#options' => $options,
-    '#default_value' => $config->get('menu_enabled') ? : [],
-    '#description' => t('Domain Access fields will be provided on each link to allow domain restrictions. Domain Menu Blocks will be provided for enabled menus.'),
-  );
-
-  $form['actions']['submit']['#submit'][] = 'domain_menu_access_form_submit';
-}
-
-function domain_menu_access_form_submit(array &$form, FormStateInterface $form_state) {
-  $config_factory = \Drupal::configFactory();
-  $config_factory->getEditable('domain_access.settings')
-    ->set('menu_enabled', $form_state->getValue('menu_enabled'))
-    ->save();
-
-  /** @var \Drupal\Core\Block\BlockManager $blockManager */
-  $blockManager = \Drupal::service('plugin.manager.block');
-  $blockManager->clearCachedDefinitions();
```

Comment:
this line is required if any menus changes -> move it to submit of settings form